src/http/common.cr

```diff
@@ -38,7 +38,7 @@ module HTTP
         if body_type.prohibited?
           body = nil
         elsif content_length = content_length(headers)
-          if content_length != 0
+          if content_length >= 0
```

Comment:
The type is `UInt64`, so this condition will always be true.

Reply:
Regardless of the type, what would it even mean to have a negative length?

Reply:
the `content_length` method seems to be calling `to_u64` on the string, so a negative length will raise overflow, I think.

I guess negative content length doesn't make sense, I was just checking because I don't know how to handle negative values. But since I now know it's handled (kind of badly) in the `content_length` method, it's probably good now.